paimon-hive/paimon-hive-connector-common/src/main/java/org/apache/paimon/hive/PaimonMetaHook.java

```diff
@@ -76,10 +77,12 @@ public void preCreateTable(Table table) throws MetaException {
 
         table.getSd().setInputFormat(PaimonInputFormat.class.getCanonicalName());
         table.getSd().setOutputFormat(PaimonOutputFormat.class.getCanonicalName());
-
-        String location = LocationKeyExtractor.getLocation(table);
+        String location = LocationKeyExtractor.getLocation(table, conf);
         if (location == null) {
             String warehouse = conf.get(HiveConf.ConfVars.METASTOREWAREHOUSE.varname);
+            org.apache.hadoop.fs.Path hadoopPath =
+                    getDnsPath(new org.apache.hadoop.fs.Path(warehouse), conf);
+            warehouse = hadoopPath.toUri().toString();
             Identifier identifier = Identifier.create(table.getDbName(), table.getTableName());
             location = AbstractCatalog.dataTableLocation(warehouse, identifier).toUri().toString();
             table.getSd().setLocation(location);
```

Comment:
Why not merge these implementation into `LocationKeyExtractor.getLocation`?

Reply:
`Identifier.class` and `AbstractCatalog.class` are used here. The two classes are defined in '`paimon-core`' module, if merge these implementation into `LocationKeyExtractor.getLocation`, we need add dependency to the `pom.xml` of the module '`paimon-hive : paimon-hive-common`' in which `LocationKeyExtractor.getLocation` is defined.